### ISSUE TITLE
Forget git-gui localization in letest releases

### ISSUE THREAD

**Issue by tonal** [git-for-windows/git#378]
In releases 2.5.1 and 2.5.2 forget localization files *.msg for git-gui and gitk-git

**Comment by dscho** [comment]
To be precise:

1. They are missing from 2.5.0 already.
2. It was a [conscious decision](https://github.com/git-for-windows/git/issues/262) to leave them out.

**Comment by tonal** [comment]
Size reduced for **portable** Git, but forget localization in regular installers Git-2.5.2.2-32-bit.exe and Git-2.5.2.2-64-bit.exe

**Comment by dscho** [comment]
> Size reduced for portable Git,

Yes.

> but forget localization

No. They were not "forgotten", they were "left out", *on purpose*.

If you want localizations back in, you *need* to convince me. Repeating that I "forgot" them, and opening new tickets for the same issue do the opposite of convincing me.

You would have to make a good case why https://github.com/git-for-windows/git/issues/183 is wrong to want English messages, and you would also have to analyze carefully how much the installers would be increased again -- just for your pleasure, and everybody else would have to wait longer to download Git for Windows and would still complain about localized messages.

In short: you need to work on this, and you need to *work* on a *convincing argument* in order to get what you want.

**Comment by rimrul** [comment]
>It was a conscious decision to leave them out.

>You would have to make a good case why #183 is wrong to want English messages, and you would also have to analyze carefully how much the installers would be increased again

How about some way to make the installer give the user the chance to opt-in to an additional download of the languages he/she is interessted in besides english? Maybe @tonal could work on that and give the users some genuine choice here without increasing the installer size too much?

**Comment by dscho** [comment]
There is an evil plan in the back of my head: once we have some proper continuous integration set up, it will be easy to auto-generate localized versions of the installers, too. So everybody can pick their poison.

However, do not hold your breath for it. If @tonal wants it so badly, why doesn't he do the same as you and I do: participate actively?

**Comment by rimrul** [comment]
Not holding any breath on it. Just pointing out an approach that might be worth conciddering.

**Comment by dscho** [comment]
> Just pointing out an approach that might be worth conciddering.

Yep, that is true.

My 2c: I'd rather not have additional downloads because then it gets harder to bundle a portable version of it, and it would also break my expectations. It would also make it a bit easier because online installers are always quite a bit more fragile than offline ones (see all the [fun](https://github.com/git-for-windows/build-extra/commit/783f742080eb8a61add0cd17bf04cbfc46f97e81) we [had](https://github.com/git-for-windows/build-extra/commit/e37f415669d053af77df7d2881f809cee8a2f190) with proxies in the SDK installer

**Comment by rimrul** [comment]
>My 2c: I'd rather not have additional downloads because then it gets harder to bundle a portable version of it, and it would also break my expectations. It would also make it a bit easier because online installers are always quite a bit more fragile than offline ones (see all the fun we had with proxies in the SDK installer

Fair point. Maybe we could do some kind of separate installation for a language pack like with .NET Framework. Maybe a self extracting 7zip with some post-install script that allows setting the new language as default.

**Comment by rimrul** [comment]
>Maybe we could do some kind of...
And by "we could do" I mean "@tonal  or whoever feels they need this could do".

**Comment by kostix** [comment]
As to web installers -- yes, they suck big time (unless, may be they are somehow integrated with BITS like that Win10 update FWIW), for instance [someone recently asked on git-users specifically for a Git installer which would come in a single piece](https://groups.google.com/d/msg/git-users/A2a15sM02AQ/gvakO24dBwAJ).

As to implementation... Would a separate `.exe` istaller containing *only* localization stuff suffice?

Alternatively, it could be possible to generate two installers dubbed "core" and "full".

**Comment by dscho** [comment]
> Would a separate .exe istaller containing only localization stuff suffice?

Probably. BTW I am no longer *that* convinced that the localizations add such a big chunk. But I refuse to do all the work when @tonal could help at least a little. This is Open Source, after all. You are empowered to help yourself, and that comes with the responsibility to try to do exactly that.

**Comment by dscho** [comment]
I think that http://www.jrsoftware.org/iskb.php?updateinstall is the way to go. If anybody wants to do more than talk about it, that is :stuck_out_tongue_winking_eye:
